# In some cases, PSIMD_SOURCE_DIR is not yet defined when configuring FP16.
# This patch ensures this is the case when FP16 is configured.
# Developed by Kurt Lust (VSC/UAntwerp)
--- pytorch-1.0.0.orig/cmake/Dependencies.cmake	2019-01-09 15:26:14.930411000 +0100
+++ pytorch-1.0.0/cmake/Dependencies.cmake	2019-01-15 11:45:58.684647267 +0100
@@ -468,6 +468,10 @@
   if (NOT DEFINED FP16_SOURCE_DIR)
     set(FP16_SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/../third_party/FP16" CACHE STRING "FP16 source directory")
   endif()
+  # Compiling FP16 also requires psimd.
+  if (NOT DEFINED PSIMD_SOURCE_DIR)
+    set(PSIMD_SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/../third_party/psimd" CACHE STRING "PSimd source directory")
+  endif()
 
   set(FP16_BUILD_TESTS OFF CACHE BOOL "")
   set(FP16_BUILD_BENCHMARKS OFF CACHE BOOL "")
